File: app/src/main/java/com/annapol04/munchkin/engine/DataType.java
package com.annapol04.munchkin.engine;

public enum DataType {
    EMPTY,
    INTEGER,
    STRING,
    CARD,
    TURN_PHASE;

    public static DataType fromId(int id) {
        if (id >= lookup.length)
            throw new IllegalArgumentException("Invalid event data type id " + id);

        return lookup[id];
    }

    private static final DataType[] lookup = new DataType[]{
            EMPTY,
            INTEGER,
            STRING,
            CARD,
            TURN_PHASE
    };
}

File: app/src/main/java/com/annapol04/munchkin/engine/DataType.kt
package com.annapol04.munchkin.engine

enum class DataType {
    EMPTY,
    INTEGER,
    STRING;

    companion object {
        private val lookup = arrayOf(EMPTY, INTEGER, STRING)

        fun fromId(id: Int): DataType {
            if (id >= lookup.size)
                throw IllegalArgumentException("Invalid event data type id $id")

            return lookup[id]
        }
    }
}